src/azure-cli/azure/cli/command_modules/role/_validators.py

```diff
@@ -84,11 +84,13 @@ def process_assignment_namespace(cmd, namespace):  # pylint: disable=unused-argu
     non_empty_msg = "usage error: {} can't be an empty string. Either omit it or provide a non-empty string."
 
     for arg in non_empty_args:
-        if getattr(namespace, arg) == "":
+        if getattr(namespace, arg, None) == "":
```

Comment:
Is there any reason to add the default value for `getattr`?

Reply:
Good question. After removing `resource_group_name` from `az role assignment create`, `resource_group_name` will not exist in `namespace`. If we don't add a default value here, `getattr` will fail for `resource_group_name` under `az role assignment create`.